src/net/ossl.cc

```diff
@@ -85,6 +87,34 @@ class ossl_error : public std::runtime_error {
     }
 };
 
+static cert_info::bytes extract_x509_serial(X509* cert) {
+    constexpr size_t serial_max = 128;
```

Comment:
nit: Per [RFC](https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.2) maximum serial number size is 20 octets